internal/endpoints/awsrulesfn/partition_test.go

```diff
@@ -0,0 +1,33 @@
+package awsrulesfn
+
+import (
+	"testing"
+)
+
+func TestGetPartition(t *testing.T) {
+	cases := map[string]struct {
+		Region        string
+		PartitionName string
+	}{
+		"test region match aws": {
+			Region: "us-west-2", PartitionName: "aws",
+		},
+		"test region match aws-cn": {
+			Region: "aws-cn-global", PartitionName: "aws-cn",
+		},
+		"test invalid region; default partition": {
+			Region: "foo", PartitionName: "aws",
+		},
+	}
+
+	for n, c := range cases {
+		t.Run(n, func(t *testing.T) {
+			p := GetPartition(partitions, c.Region)
```

Comment:
**Suggestion**: This is probably ok because the partitions don't change often but this test would be more immune to external partition changes if the test was against a mock list of partitions rather than the generated one. A change to `partitions.json` could theoretically break this. Perhaps it's unlikely enough to not matter though.

Reply:
I agree, unit tests should be more isolated to external changes. I can make this change since it shouldnt be too heavy of a lift.